hyperspy/drawing/image.py

```diff
@@ -532,8 +541,11 @@ def format_coord(x, y):
         if self.no_nans:
             data = np.nan_to_num(data)
 
-        if ims:  # the images has already been drawn previously
-            ims[0].set_data(data)
+        if ims:  # the images have already been drawn previously
+            if self.mesh:
+                self.mesh.update({'array': data.ravel()})
+            else:
+                ims[0].set_data(data)
```

Comment:
```suggestion
            if self.mesh:
                self.mesh.set_array(data.ravel())
            else:
                ims[0].set_data(data)
```
OR
```suggestion
            if self.mesh:
                self.mesh.update({'array': data.ravel()})
            else:
                ims[0].update({'data'}: data)
```

From reading matplotlib code, I am pretty sure that both do the same thing... https://github.com/matplotlib/matplotlib/blob/d5210e4a87665581205f5147703dbd229b44cf66/lib/matplotlib/artist.py#L1205-L1231 and just to make more readable, it would be good to keep only one of the two.

Reply:
I decided to use the first option.